src/cli/ofp_cli_arp.c

```diff
@@ -47,13 +47,62 @@ void f_arp_cleanup(struct cli_conn *conn, const char *s)
 	sendcrlf(conn);
 }
 
+void f_arp_add(struct cli_conn *conn, const char *s)
+{
+	int a, b, c, d, e, f, g, h, i, j;
+	char dev[16];
+	int port, vlan;
+	struct ofp_ifnet *itf = NULL;
+	uint32_t ipv4_addr;
+	uint8_t mac[OFP_ETHER_ADDR_LEN];;
+
+	if (sscanf(s, " %d.%d.%d.%d %x:%x:%x:%x:%x:%x %s",
+		&a, &b, &c, &d, &e, &f, &g, &h, &i, &j, dev) != 11)
+		return;
+
+	ipv4_addr = odp_cpu_to_be_32((a << 24) | (b << 16) | (c << 8) | d);
+	mac[0] = e;mac[1] = f;mac[2]=g;
+	mac[3] = h;mac[4] = i;mac[5]=j;
+
+	port = ofp_name_to_port_vlan(dev, &vlan);
+	if (port == -1 || !PHYS_PORT(port)) {
+		ofp_sendf(conn->fd, "Invalid device name.\r\n");
+		sendcrlf(conn);
+		return;
+	}
+
+	itf = ofp_get_ifnet(port, vlan);
+	if (itf == NULL) {
+		ofp_sendf(conn->fd, "Device not found.\r\n");
+		sendcrlf(conn);
+		return;
+	}
+
+	if (ofp_arp_ipv4_insert(ipv4_addr, mac, itf)) {
+		ofp_sendf(conn->fd, "Failed to insert arp entry.\r\n");
+		sendcrlf(conn);
+		return;
+	}
+
+	sendcrlf(conn);
+}
+
 void f_help_arp(struct cli_conn *conn, const char *s)
 {
 	(void)s;
 	ofp_sendf(conn->fd,
 		"Show arp table:\r\n"
 		"  arp\r\n\r\n");
 
+	ofp_sendf(conn->fd,
+		"Add arp entry:\r\n"
+		"  arp set IP4ADDR HWADDR dev DEV\r\n"
+		"    IP4ADDR: ip address in a.b.c.d format\r\n"
+		"    HWADDR: mac address\r\n"
```

Comment:
For uniformity you could add the address format also for HWADDR or remove the format from IP4ADDR (the format is already show in the example).